ark/analysis/spatial_analysis.py

```diff
@@ -294,3 +294,65 @@ def create_neighborhood_matrix(all_data, dist_matrices_dict, included_fovs=None,
         cell_neighbor_freqs.loc[current_fov_neighborhood_data.index, fov_cluster_names] = freqs
 
     return cell_neighbor_counts, cell_neighbor_freqs
+
+
+def cluster_neighborhood_matrix(neighbor_mat, max_k=10, included_fovs=None, fov_col='SampleID',
```

Comment:
This seems more like the name for the actual function which performs k-means clustering on the data, not the function used to determine the right score.